File: back/src/main/java/com/back/global/exception/ServiceException.java
package com.back.global.exception;

import com.back.global.rsData.RsData;
import lombok.Getter;

@Getter
public class ServiceException extends RuntimeException {

    private final ErrorCode errorCode;

    public ServiceException(ErrorCode errorCode) {
        super(errorCode.getMessage());
        this.errorCode = errorCode;
    }

    public RsData<Void> getRsData() {
        return new RsData<>(
                errorCode.getResultCode(),
                errorCode.getMessage(),
                null
        );
    }
}

File: back/src/main/kotlin/com/back/global/exception/ServiceException.kt
package com.back.global.exception

import com.back.global.rsData.RsData

class ServiceException(
    val errorCode: ErrorCode
) : RuntimeException(errorCode.message) {

    val rsData: RsData<Void>
        get() = RsData(
            resultCode = errorCode.resultCode,
            msg = errorCode.message,
            data = null
        )
}
